pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -580,45 +638,37 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListenerForPortOrUDS(no
 		switch pluginParams.ListenerProtocol {
 		case plugin.ListenerProtocolHTTP:
 			filterChainMatch = chain.FilterChainMatch
+			// This is the filter chain used by permissive mTLS. Append applicationProtocolsOverTLS as the client side will
+			// override the ALPN with applicationProtocolsOverTLS.
+			if filterChainMatch != nil && len(filterChainMatch.ApplicationProtocols) > 0 {
+				filterChainMatch.ApplicationProtocols = append(filterChainMatch.ApplicationProtocols, applicationProtocolsOverTLS...)
```

Comment:
Since this is pure http case, why add this?

Reply:
For strict mTLS

Reply:
The ALPN filter override the ALPN to ["istio-http/1.1"...], we need to append the ALPN in the inbound listener.